### ISSUE TITLE
Setting "0.0.0.0" for host configuration in config/vite.json is not working correctly

### ISSUE THREAD

**Issue by vtt** [ElMassimo/vite_ruby#179]
### Description 📖

When specifying "0.0.0.0" as host in config/vite.json, HMR is not working correctly.
config/vite.json
```
{
  "all": {
    "sourceCodeDir": "app/frontend",
    "watchAdditionalPaths": []
  },
  "development": {
    "autoBuild": true,
    "host": "0.0.0.0",
    "publicOutputDir": "vite_dev",
    "port": 3036
  },
  "test": {
    "autoBuild": true,
    "publicOutputDir": "vite_test",
    "port": 3037
  }
}
```

Running `bin/vite dev`, it shows: 

```
vite v2.7.12 dev server running at:

  > Local:    http://localhost:3036/vite_dev/
  > Network:  http://192.168.0.4:3036/vite_dev/

  ready in 467ms.

```

Under Chrome console, it shows:
```
[vite] connecting...
client.ts:28 WebSocket connection to 'ws://0.0.0.0:3036/vite_dev/' failed: 
(anonymous) @ client.ts:28
client.ts:222 [vite] server connection lost. polling for restart...
```

Note that it uses "0.0.0.0" as the address but not the correct IP for the network 192.168.0.4

**Comment by ElMassimo** [comment]
See https://github.com/ElMassimo/vite_ruby/pull/119#issuecomment-1013521334 and leave your feedback, to consider reverting that behavior.

In the meantime, you can configure `hmr.host` explicitly in`vite.config.ts`.

**Comment by ElMassimo** [comment]
Released [`vite-plugin-ruby@3.0.8`](https://github.com/ElMassimo/vite_ruby/blob/main/vite-plugin-ruby/CHANGELOG.md#308-2022-01-18) including this fix.

**Comment by wizztjh** [comment]
I'm still having this issue in Vite v3.1.7

```
VITE v3.1.7  ready in 1269 ms
hyonline_internal-web-1  | 07:05:30 vite.1 |
hyonline_internal-web-1  | 07:05:30 vite.1 |   ➜  Local:   http://localhost:3036/vite/
hyonline_internal-web-1  | 07:05:30 vite.1 |   ➜  Network: http://172.18.0.3:3036/vite/
```

![image](https://user-images.githubusercontent.com/462417/195019834-50b50212-15b9-453f-a713-1dd99f24acf0.png)
![image](https://user-images.githubusercontent.com/462417/195019944-48f72a91-76e6-4f95-ba4f-5cee63e18287.png)
